lib/active_model_serializers/adapter/base.rb

```diff
@@ -74,7 +74,8 @@ def fragment_cache(cached_hash, non_cached_hash)
       # see https://github.com/rails-api/active_model_serializers/pull/965
       # When <tt>options</tt> is +nil+, sets it to +{}+
       def serialization_options(options)
-        options ||= {} # rubocop:disable Lint/UselessAssignment
+        options ||= {}
+        options.frozen? ? options.dup : options
```

Comment:
Curious if the cost of checking for freezing here is worth it vs. always duplicating.

Reply:
I ran a synthetic benchmark, and both options show almost identical performance:

```
                             user     system      total        real
Always dup:              0.458805   0.015979   0.474784 (  0.475676)
Cond. dup (frozen):      0.460826   0.014778   0.475604 (  0.476971)
Cond. dup (not frozen):  0.459270   0.010303   0.469573 (  0.471081)
```

For reference, [here is the script I used](https://gist.github.com/pulkit110/08d963ddffce8be4c87e0efb03f1d048).